fabric/src/test/java/TestUnsignedByte.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+import com.wynntils.utils.type.UnsignedByte;
+import net.minecraft.SharedConstants;
+import net.minecraft.server.Bootstrap;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class TestUnsignedByte {
+    @BeforeAll
+    public static void setup() {
+        SharedConstants.tryDetectVersion();
+        Bootstrap.bootStrap();
+    }
+
+    @Test
+    public void simpleByteArray_toUnsignedByteWorks() {
+        UnsignedByte[] bytes = UnsignedByte.of(new byte[] {(byte) 214, 121, 11, 49, 43, 75});
```

Comment:
It's kind of only the first value that actually tests that the unsigned handling works... :)

What would be the expected outcome of sending in a negative valued byte? Should it be tested?

Reply:
214 is basically a negative byte though. It would be converted into it's unsigned representation. But negative bytes are not used by the encoding, only unsigned bytes 0-255.

Reply:
Yes, that is why I wrote that it is only 214 that is doing any real testing. I should clarify: I think you should add more "negative" bytes, and especially the borderline cases like 127, 128, 0, 1. 

And my second question was: what would be the result of UnsignedByte.of(new byte[] { -1, -2, -127, -128, -129 })? I think it would be a useful test case to add.

Reply:
Added some test cases for them :)